src/andromede/model/constraint.py

```diff
@@ -32,55 +35,39 @@ class Constraint:
 
     name: str
     expression: ExpressionNode
-    lower_bound: ExpressionNode
-    upper_bound: ExpressionNode
-    context: ProblemContext
+    lower_bound: ExpressionNode = field(default=literal(-float("inf")))
+    upper_bound: ExpressionNode = field(default=literal(float("inf")))
+    context: ProblemContext = field(default=ProblemContext.OPERATIONAL)
 
-    def __init__(
+    def __post_init__(
         self,
-        name: str,
-        expression: ExpressionNode,
-        lower_bound: Optional[ExpressionNode] = None,
-        upper_bound: Optional[ExpressionNode] = None,
-        context: ProblemContext = ProblemContext.OPERATIONAL,
     ) -> None:
-        self.name = name
-        self.context = context
-
-        if isinstance(expression, ComparisonNode):
-            if lower_bound is not None or upper_bound is not None:
+        if isinstance(self.expression, ComparisonNode):
+            if not is_unbound(self.lower_bound) or not is_unbound(self.upper_bound):
```

Comment:
For the ComparisonNode part, it won't be a problem because some lines below we set the bounds ourselves.
However, for LiteralNodes and ParameterNodes (else statement), I agree with you. I will fix it